src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -146,7 +147,10 @@ public IEqualityComparer<TKey> Comparer
         {
             get
             {
-                return (_comparer == null || _comparer is NonRandomizedStringEqualityComparer) ? EqualityComparer<TKey>.Default : _comparer;
+                IEqualityComparer<TKey> comparer = _comparer;
+                // Use public randomized comparer alternative for Comparer property (if non-randomized version is in use)
+                HashHelpers.TrySwitchRandomizedHashCodeGenerator(ref comparer);
```

Comment:
If we are using `Try*` semantics/style then we should check the return and branch on that. Otherwise, just return the result and keep the logic the same?